chore: Remove unused cloud sync related files

nj1973 opened the pull request.

abb9979 commented:
Some references to the Cloud Sync remote_offload.conf file:

* goe/src/goe/util/misc_functions.py: `check_remote_offload_env()` function
* goe/templates/conf/offload.env.template.common: `export REMOTE_OFFLOAD_CONF=...`

nj1973 commented:
> Some references to the Cloud Sync remote_offload.conf file:
> 
> * goe/src/goe/util/misc_functions.py: `check_remote_offload_env()` function
> * goe/templates/conf/offload.env.template.common: `export REMOTE_OFFLOAD_CONF=...`

I've removed these, thanks for the reminder. I also removed a few other functions that were no longer used.

All unit and integration tests pass.
